orion-commons/src/main/java/com/pinterest/orion/utils/HttpDeleteWithBody.java

```diff
@@ -0,0 +1,23 @@
+package com.pinterest.orion.utils;
+
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+
+import java.net.URI;
+
+/**
+ * Extends HttpEntityEnclosingRequestBase to allow for a body in a DELETE request.
+ * HttpDelete does not allow for a body in the request.
+ * This class is necessary to allow for a body in a DELETE request.
+ */
+class HttpDeleteWithBody extends HttpEntityEnclosingRequestBase {
+    public static final String METHOD_NAME = "DELETE";
+
+    public String getMethod() {
+        return METHOD_NAME;
+    }
+
+    public HttpDeleteWithBody(final String uri) {
```

Comment:
No. That's the target URI of this HTTP request. 

For regular HttpDelete request, I cannot run setEntity and attach the instance id to it.

With this HttpDeleteWithBody class, it will be executed as HttpDelete request by the http client because of the getMethod value, and extends HttpEntityEnclosingRequestBase allows me to attach body (entity) to it.